packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -271,3 +281,76 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   componentId?: string;
 }
+
+export type ITextFieldStyleProps = Required<Pick<ITextFieldProps, 'theme'>> &
+  Pick<
+    ITextFieldProps,
+    'className' | 'disabled' | 'required' | 'multiline' | 'borderless' | 'resizable' | 'underlined'
+  > & {
+    /** Element has an error message. */
+    hasErrorMessage?: boolean;
+    /** Element has an icon. */
+    hasIcon?: boolean;
+    /** Element has a label. */
+    hasLabel?: boolean;
+    /** Element has focus. */
+    focused?: boolean;
+  };
+
+export interface ITextFieldSubComponentStyles {
+  // TODO: this should be the interface once we're on TS 2.9.2 but otherwise causes errors in 2.8.4
+  // label: IStyleFunctionOrObject<ILabelStyleProps, ILabelStyles>;
+  label: IStyleFunctionOrObject<any, any>;
+}
```

Comment:
Hmm it's actually interesting you made this a dedicated interface. This way the autodoc picks it up - but somewhat ironically the label itself is missing JSDoc :)